src/js/components/Diagram/Diagram.js

```diff
@@ -222,20 +273,49 @@ const Diagram = forwardRef(({ connections, ...rest }, ref) => {
             colorName = colors[index % colors.length];
           }
 
+          let arrowMarker = null;
+
+          if (typeof endpoint === 'object' && endpoint?.from === 'arrow') {
+            arrowMarker = openArrow(
+              normalizeColor(colorName, theme),
+              index,
+              'openArrowStart',
```

Comment:
also to be more flexible for future, should we name this `endpointStart` since it might not always be an arrow?